File: 1291-Sequential_Digits/Sequential_Digits.go
package sequentialdigits

func sequentialDigits(low int, high int) []int {
	answer := make([]int, 0)

	lowDigitCount := 0
	highDigitCount := 0

	currLow := low
	currHigh := high

	for currLow > 0 {
		currLow /= 10
		lowDigitCount++
	}

	for currHigh > 0 {
		currHigh /= 10
		highDigitCount++
	}

	for digitCount := lowDigitCount; digitCount <= highDigitCount; digitCount++ {
		for curr := 1; curr <= 10-digitCount; curr++ {
			num := curr
			for nextNum := curr + 1; nextNum < curr+digitCount; nextNum++ {
				num *= 10
				num += nextNum
			}

			if num < low {
				continue
			} else if num <= high {
				answer = append(answer, num)
			} else {
				break
			}

		}
	}

	return answer
}

File: 1291-Sequential_Digits/Sequential_Digits.java
class Solution {
    List<Integer> answer = new ArrayList<>();
    public List<Integer> sequentialDigits(int low, int high) {
        int highDigitCount = 0;
        int lowDigitCount = 0;
        
        int currentLow = low;
        int currentHigh = high;
        
        while(currentLow > 0){
            currentLow /= 10;
            lowDigitCount++;
        }
        
        while(currentHigh > 0){
            currentHigh /= 10;
            highDigitCount++;
        }
        
        for(int digitCount = lowDigitCount; digitCount <= highDigitCount; digitCount++){
            generateSequentialDigits(low, high, digitCount);
        }
        
        return answer;
    }
    
    private void generateSequentialDigits(int low, int high, int digitCount){
        for(int startDigit = 1; startDigit <= 10 - digitCount; startDigit++){
            int num = startDigit;
            for(int nextDigit = 1; nextDigit < digitCount ; nextDigit++){
                num *= 10;
                num += (startDigit + nextDigit);
            }
            if (num < low) continue;
            else if(num <= high) answer.add(num);
            else return;
        }
    }
}